beancount_import/matching.py

```diff
@@ -240,6 +258,28 @@ def get_fuzzy_date_range(self, orig_date: datetime.date):
                                 self.fuzzy_match_days + 1):
             yield orig_date + datetime.timedelta(days=day_offset)
 
+    def fuzz_date_currency_key(self, key: DateCurrencyKey) -> Iterable[DateCurrencyKey]:
+        for d in self.get_fuzzy_date_range(key[0]):
+            yield d, key[1]
+
+    def _date_currency_key(self, entry: Transaction, mp: MatchablePosting) -> DateCurrencyKey:
+        pw = get_posting_weight(mp.posting)
+        currency = ""
+        if pw:
+            currency = pw.currency
+        return (_date_key(entry, mp), currency)
+
+    def _search_posting(self, entry: Transaction, mp: MatchablePosting):
+        pw = get_posting_weight(mp.posting)
+        number = None
```

Comment:
number should also always be present for a matchable posting

Reply:
See https://github.com/jbms/beancount-import/pull/174#discussion_r1089647813